From f0d510468f97af916a284a17900f07a476d6b7dc Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:12:31 -0400
Subject: [powerpc] mm: use macro PGTABLE_EADDR_SIZE instead of digital

Message-id: <1410545655-205645-523-git-send-email-dzickus@redhat.com>
Patchwork-id: 94512
O-Subject: [RHEL7 PATCH 522/626] powerpc/mm: use macro PGTABLE_EADDR_SIZE instead of digital
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 5a4e58bc693f04aa650219784e5e339e0db6b902
Author: Liu Ping Fan <kernelfans@gmail.com>
Date:   Thu Nov 21 10:17:54 2013 +0800

    powerpc/mm: use macro PGTABLE_EADDR_SIZE instead of digital

    In case of extending the eaddr in future, use this macro
    PGTABLE_EADDR_SIZE to ease the maintenance of the code.

    Signed-off-by: Liu Ping Fan <pingfank@linux.vnet.ibm.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/mm/slb_low.S b/arch/powerpc/mm/slb_low.S
index 28cffb6..736d18b 100644
--- a/arch/powerpc/mm/slb_low.S
+++ b/arch/powerpc/mm/slb_low.S
@@ -35,7 +35,7 @@ _GLOBAL(slb_allocate_realmode)
   * check for bad kernel/user address
   * (ea & ~REGION_MASK) >= PGTABLE_RANGE
   */
- rldicr. r9,r3,4,(63 - 46 - 4)
+ rldicr. r9,r3,4,(63 - PGTABLE_EADDR_SIZE - 4)
  bne- 8f
 
  srdi r9,r3,60  /* get region */
-- 
1.7.1